src/managers/AutoModerationRuleManager.js

```diff
@@ -0,0 +1,293 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const CachedManager = require('./CachedManager');
+const AutoModerationRule = require('../structures/AutoModerationRule');
+const {
+  AutoModerationRuleEventTypes,
+  AutoModerationRuleTriggerTypes,
+  AutoModerationActionTypes,
+  AutoModerationRuleKeywordPresetTypes,
+} = require('../util/Constants');
+
+/**
+ * Manages API methods for auto moderation rules and stores their cache.
+ * @extends {CachedManager}
+ */
+class AutoModerationRuleManager extends CachedManager {
+  constructor(guild, iterable) {
+    super(guild.client, AutoModerationRule, iterable);
+
+    /**
+     * The guild this manager belongs to.
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  _add(data, cache) {
+    return super._add(data, cache, { extras: [this.guild] });
+  }
+
+  /**
+   * Options used to set the trigger metadata of an auto moderation rule.
+   * @typedef {Object} AutoModerationTriggerMetadataOptions
+   * @property {string[]} [keywordFilter] The substrings that will be searched for in the content
+   * @property {(string|RegExp)[]} [regexPatterns] The regular expression patterns
```

Comment:
This should not accept RegExp instances
https://github.com/discordjs/discord.js/pull/7938#discussion_r1020837942